testutils/clustermanager/gke.go

```diff
@@ -260,13 +261,70 @@ func (gc *GKECluster) Acquire() error {
 	return err
 }
 
-// Delete deletes a GKE cluster
-func (gc *GKECluster) Delete() error {
-	if !gc.NeedCleanup {
-		return nil
+// wait depends on unique opName(operation ID created by cloud), and waits until
+// it's done
+func (gc *GKECluster) wait(location, opName string, wait time.Duration) error {
+	validStatuses := []string{"PENDING", "RUNNING", "DONE"}
+	doneStatus := "DONE"
+	var op *container.Operation
+	var err error
+
+	timeout := time.After(wait)
+	tick := time.Tick(500 * time.Millisecond)
+	for {
+		select {
+		// Got a timeout! fail with a timeout error
+		case <-timeout:
+			return fmt.Errorf("timed out waiting")
+		case <-tick:
+			// Retry 3 times in case of weird network error, or rate limiting
+			r := 0
+			for w := 50 * time.Microsecond; r < 3; w *= 2 {
+				r++
+				op, err = gc.operations.getOperation(*gc.Project, location, opName)
+				if nil == err {
+					valid := false
+					for _, validStatus := range validStatuses {
+						if op.Status == validStatus {
+							valid = true
+						}
+					}
+					if !valid {
+						err = fmt.Errorf("unexpected operation status: '%s'", op.Status)
```

Comment:
Is there any reason to continue looping?
Also if first iteration succeeds but the second one fails, `err` will be non-nil.

Reply:
I have seen it once there was an intermittent wrong state during waiting, but it fixed itself, comment added. If the first iteration succeeded it shouldn't go to second one, PTAL.